wrangler-core/src/main/java/io/cdap/wrangler/utils/ArithmeticOperations.java

```diff
@@ -388,12 +390,41 @@ public static Float min(Float f1, Float f2) {
   }
 
   /**
-   Arithmetic operation - Find the minimum of two columns.
+   * Arithmetic operation - Find the minimum of two columns.
    */
   public static BigDecimal min(BigDecimal bd1, BigDecimal bd2) {
     if (bd1 == null || bd2 == null) {
       return null;
     }
     return bd1.min(bd2);
   }
+
+  /**
+   * Arithmetic operation - calculate average of any number of columns.
+   * Will throw an error if any input is of type BigDecimal or Byte.
+   */
+  public static Double average(Number... nums) throws DirectiveExecutionException {
+    // Note that this algorithm avoids overflow
+    double avg = 0;
+    int t = 1;
+    boolean allNull = true;
+    for (Number num : nums) {
+      // Check for null input
+      if (num == null) {
+        continue;
+      } else {
+        allNull = false;
+      }
+      // Check for invalid input
+      if (num instanceof BigDecimal || num instanceof Byte) {
+        throw new DirectiveExecutionException("Input cannot be of type 'Decimal' or 'Byte'.");
```

Comment:
Is there any reason why we decided to skip Decimal support from this function?

I understand there may be precision loss when casting to Double, but it seems arbitrary to not support the average function on Decimal columns